src/main/java/LinkedBinarySearchTree.java

```diff
@@ -13,7 +13,7 @@ public LinkedBinarySearchTree() {
 
     public LinkedBinarySearchTree(T element) {
         root = new Node<>(element);
-        size = 0;
+        size = 1;
     }
```

Comment:
May want to consider writing this as:
```java
public LinkedBinarySearchTree(T element) {
    this();
    add(element);
}
```

It's usually good practice to have constructors call `this()` (to other ctors in general, except the main one getting delegated to). It's not usually wise to call non-final methods in the constructor (though, [java does it](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/LinkedList.java#L113-L124)), but it's also weird to duplicate the `add` logic here. In reality, I don't think you'd see this constructor added to the class in the first place. Cloning entire collections, sure, but adding a single element is not much reason to add a new constructor. For learning purposes, it's fine though.

Reply:
Good call. In fact, not sure why I didn't do it that way since I did it before 🤷 Will also add a point for it being perhaps unnecessary.